pkg/apps/application-list.jsx

```diff
@@ -117,14 +122,35 @@ export const ApplicationList = ({ metainfo_db, appProgress, appProgressTitle, ac
         }
     }
 
+    async function check_missing_data(packages) {
+        try {
+            const missing = await check_uninstalled_packages(packages);
+            setDataPackagesInstalled(missing.size === 0);
+        } catch (e) {
+            console.warn("Failed to check missing AppStream metadata packages:", e.toString());
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19281-20231012-120844-1a910def-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)